src/Adapter/Sftp.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Bolt\Filesystem\Adapter;
+
+use League\Flysystem\Config;
+use League\Flysystem\Sftp\SftpAdapter;
+
+/**
+ * SFTP adapter.
+ *
+ * @author Gawain Lynch <gawain.lynch@gmail.com>
+ */
+class Sftp extends SftpAdapter
+{
+    /** @var int */
+    protected $filePerm = 0644;
+    /** @var int */
+    protected $directoryPerm = 0755;
+
+    /**
+     * @inheritdoc
+     */
+    public function __construct(array $config)
+    {
+        $this->configurable[] = 'filePerm';
+        parent::__construct($config);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function has($path)
+    {
+        $location = $this->applyPathPrefix($path);
+
+        return (bool) $this->getMetadata($location);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getMetadata($path)
+    {
+        $metadata = parent::getMetadata($path);
+        if (is_array($metadata)) {
+            $metadata += ['path' => $path];
+        }
+
+        return $metadata;
+    }
```

Comment:
Ah I see that our `Filesystem::getTypeFromMetadata` needs it. That's a bug that should be fixed there instead.

```php
// Right before getTypeFromMetadata call in doGetType method.
$metadata += ['path' => $path];
```